source/default/_types/particle-element.ts

```diff
@@ -0,0 +1,13 @@
+export default class ParticleElement {
```

Comment:
I think we could also create a ParticleInterface which this class implements to keep all of the type definitions concise and in the interface. It could then also include the types of the arguments that the various functions utilize.

Reply:
Agreed :-) I'll add that